packages/core/dist/GondelComponent.d.ts

```diff
@@ -1,4 +1,11 @@
-export declare type IGondelComponent = new (context: HTMLElement, componentName: string) => GondelComponent;
+export interface IGondelComponent<T = GondelComponent> {
```

Comment:
If we add the static variable to `GondelBaseComponent` we could switch to `T extends GondelBaseComponent = GondelBaseComponent` so we would no longer need IGondelComponentWithIdentification.